bites_test.py

```diff
@@ -596,5 +597,31 @@ def __init__(self, name, score):
     self.assertEqual(print_mock.call_args_list[1], mock.call("luigi: 9\n"))
     print_patcher.stop()
 
+class PlayFullGameTest(unittest.TestCase):
+  @patch('bites.Bites.take_all_turns')
+  def test_play_full_game_calls_take_all_turns(self, mock):
+  # test 82
+    bites_game = Bites([], {}, [])
+    bites_game.play_full_game()
+    self.assertTrue(mock.called)
+
+  @patch('bites.Bites.print_scores')
+  def test_play_full_game_calls_print_scores(self, mock):
+  # test 83
+    class FakePlayer():
+      def __init__(self, name, score):
+        self.name = name
+        self.score = score
+      
+      def take_turn(self, trail, ant_positions, anthill):
+        return (trail, ant_positions, anthill)
+
+    fake_mario = FakePlayer("mario", 3)
+    fake_luigi = FakePlayer("luigi", 9)
+    players = [fake_mario, fake_luigi]
```

Comment:
Do you really need the fake players, considering you've mocked the score printing? I would expect not. I would expect that an empty player list would still work, like it does in test 82.

Reply:
I don't know the exact reason, but no; it does not work with an empty list. This was partly what took a long time to resolve. When run with an empty list in lieu of players; the test gets stuck in an infinite loop and I have to Ctrl+C. My hunch is that print_scores() needs an object with self.name and self.score attributes, hence the FakePlayer class being used.